cli/cmd/graphenePrepare.go

```diff
@@ -0,0 +1,391 @@
+	backupFileName := filepath.Base(fileName) + ".bak"
+	fmt.Printf("Saving original manifest as %s...\n", backupFileName)
+	if err := ioutil.WriteFile(filepath.Join(directory, backupFileName), manifestContentOriginal, 0644); err != nil {
+		return err
+	}
+
+	// Write modified file to disk
+	fileNameBase := filepath.Base(fileName)
+	fmt.Printf("Saving changes to %s...\n", fileNameBase)
+	if err := ioutil.WriteFile(fileName, manifestContentModified, 0644); err != nil {
+		return err
+	}
+
+	fmt.Println("Downloading Marblerun premain from GitHub...")
+	// Download Marblerun premain for Graphene from GitHub
+	if err := downloadPremain(directory, mode); err != nil {
+		var fileName string
+		if mode == modeSpawn {
+			fileName = premainNameSpawn
+		} else if mode == modePreload {
+			fileName = premainNamePreload
+		}
+		color.Red("ERROR: Cannot download '%s' from GitHub. Please add the file manually.", fileName)
+	}
+
+	fmt.Println("\nDone! You should be good to go for Marblerun!")
+
+	return nil
+}
+
+func downloadPremain(directory string, mode mode) error {
+	cleanVersion := "v" + strings.Split(Version, "-")[0]
+
+	// Download premain-graphene as executable (spawn) or as shared library (preload), depending on user's choice
+	var downloadName string
+	if mode == modeSpawn {
+		downloadName = premainNameSpawn
+	} else if mode == modePreload {
+		downloadName = premainNamePreload
+	} else {
+		return errors.New("unknown premain mode, cannot download premain")
+	}
+
+	resp, err := http.Get(fmt.Sprintf("https://github.com/edgelesssys/marblerun/releases/download/%s/%s", cleanVersion, downloadName))
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return errors.New("received a non-successful HTTP response")
+	}
+
+	out, err := os.Create(filepath.Join(directory, downloadName))
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	if _, err := io.Copy(out, resp.Body); err != nil {
+		return err
+	}
+
+	fmt.Printf("Successfully downloaded %s.\n", downloadName)
+
+	return nil
+}
+
+/*
+	Perform the manifest modification.
+	For existing entries: Run a RegEx search, replace the line.
+	For new entries: Append to the end of the file.
+	NOTE: This only works for flat-mapped TOML configs.
+	These seem to be usually used for Graphene manifests.
+	However, TOML is quite flexible, and there are no TOML parsers out there which are style & comments preserving
+	So, if we do not have a flat-mapped config, this will fail at some point.
+*/
+func appendAndReplace(changeDiffs []diff, manifestContent []byte) ([]byte, error) {
+	newManifestContent := manifestContent
+
+	var firstAdditionDone bool
+	for _, value := range changeDiffs {
+		if value.alreadyExists {
+			// If a value was previously existing, we replace the existing entry
+			key := strings.Split(value.manifestEntry, " =")
+			regexKey := strings.ReplaceAll(key[0], ".", "\\.")
+			regex := regexp.MustCompile("(?m)^" + regexKey + "\\s?=.*$")
```

Comment:
```suggestion
			regex := regexp.MustCompile("(?m)^\\s*" + regexKey + "\\s*=.*$")
```
should we accept arbitrary whitespaces here?

Reply:
I think it's not bad to support it. Might help when a manifest is slightly misformatted (douple spaces, lack of spaces etc.), but we're still sure that this is the right line to replace.

Reply:
Might help for badly formatted files where we still know the exact line to replace. Any ideas where this could go wrong and cause mismatches?

Otherwise we could limit it to a certain number, though I think that would be an odd solution.

Reply:
Don't see mismatches, but am not confident. Decide yourself to change or keep as is.